### ISSUE TITLE
The box_loss is zero

### ISSUE THREAD

**Issue by Justmore5mins** [ultralytics/ultralytics#22228]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

All the other epochs like the epoch shown but the cls_loss floating, I can't figure out why the box_loss is always zero

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      10/10      6.23G          0      64.14          0          8        640: 100% ━━━━━━━━━━━━ 13/13 0.9it/s 13.8s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 2/2 0.7it/s 3.0s
                   all         50         46          0          0          0          0

(I've asked the Warp integrated AI model and still not able to solve the problem

### Environment

yolo check
Ultralytics 8.3.203 🚀 Python-3.13.7 torch-2.10.0.dev20250926 CPU (Apple M3 Pro)
Setup complete ✅ (11 CPUs, 18.0 GB RAM, 273.6/460.4 GB disk)

OS                     macOS-26.0-arm64-arm-64bit-Mach-O
Environment            Darwin
Python                 3.13.7
Install                pip
Path                   /Users/justmore5mins/Documents/documents/code/AIDeployment/venv/lib/python3.13/site-packages/ultralytics
RAM                    18.00 GB
Disk                   273.6/460.4 GB
CPU                    Apple M3 Pro
CPU count              11
GPU                    None
GPU count              None
CUDA                   None

numpy                  ✅ 2.2.6>=1.23.0
matplotlib             ✅ 3.10.6>=3.3.0
opencv-python          ✅ 4.12.0.88>=4.6.0
pillow                 ✅ 11.3.0>=7.1.2
pyyaml                 ✅ 6.0.2>=5.3.1
requests               ✅ 2.32.5>=2.23.0
scipy                  ✅ 1.16.2>=1.4.1
torch                  ✅ 2.10.0.dev20250926>=1.8.0
torch                  ✅ 2.10.0.dev20250926!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.25.0.dev20250926>=0.9.0
psutil                 ✅ 7.1.0
polars                 ✅ 1.33.1
ultralytics-thop       ✅ 2.0.17>=2.0.0

### Minimal Reproducible Example

I use cli to train the model for debugging like

    yolo detect train model="yolo12n.pt" data="datasets/cube/data.yaml" epochs=10 device=mps plots=False

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Justmore5mins, thanks for opening this issue and for using Ultralytics 🚀 This is an automated response to help you get support faster. An Ultralytics engineer will follow up shortly to assist you further.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

Since you’re seeing `box_loss` as zero during training, please share the following details to help us reproduce and diagnose:
- Exact command used (you shared one—please also include any prior setup commands and the full console output from start to finish)  
- A small sample of your dataset (a few images with their corresponding label .txt files)  
- Your `datasets/cube/data.yaml` contents (classes/names, paths)  
- A screenshot or snippet of one label file to confirm format  
- Any custom settings (imgsz, batch, workers, seed, augmentations)  
- Model file and version used (e.g., `yolo12n.pt`) and `ultralytics` package version (`yolo version` or `pip show ultralytics`)  
- Brief confirmation whether labels are non-empty and match the declared classes in `data.yaml`  
- If possible, try a quick run with `device=cpu` and share if behavior differs from `device=mps`  
- Validation command/output you used to check labels (if any)

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). 🙌

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You can try

```
pip uninstall ultralytics -y && pip install git+https://github.com/ultralytics/ultralytics
```

**Comment by Justmore5mins** [comment]
ohhh thank you so much (I don't thought that its solved that easily)

**Comment by glenn-jocher** [comment]
Great to hear it’s resolved—quick tip: for stable training please use YOLO11 (YOLO12 isn’t recommended) and, on Apple Silicon, verify MPS before training:

```python
import torch
print(torch.backends.mps.is_available())
```

Then run:

```bash
yolo detect train model=yolo11n.pt data=datasets/cube/data.yaml device=mps
```

If this resurfaces, open a new issue with `yolo checks` output and full logs; references: [YOLO11 models](https://docs.ultralytics.com/models/yolo11/) and [Troubleshooting Common YOLO Issues](https://docs.ultralytics.com/guides/yolo-common-issues/).
